habitat_baselines/common/obs_transformers.py

```diff
@@ -0,0 +1,630 @@
+    def from_config(cls, config):
+        return cls(
+            size=config.RL.POLICY.OBS_TRANSFORMS.RESIZE_SHORTEST_EDGE.SIZE
+        )
+
+
+@baseline_registry.register_obs_transformer()
+class CenterCropper(ObservationTransformer):
+    def __init__(
+        self,
+        size: Union[int, Tuple[int]],
+        channels_last: bool = True,
+        trans_keys: Tuple[str] = ("rgb", "depth", "semantic"),
+    ):
+        r"""An nn module that center crops your input.
+        Args:
+            size: A sequence (h, w) or int of the size you wish to resize/center_crop.
+                    If int, assumes square crop
+            channels_list: indicates if channels is the last dimension
+        """
+        super().__init__()
+        if isinstance(size, numbers.Number):
+            size = (int(size), int(size))
+        assert len(size) == 2, "forced input size must be len of 2 (h, w)"
+        self._size = size
+        self.channels_last = channels_last
+        self.trans_keys = trans_keys
+
+    def transform_observation_space(
+        self,
+        observation_space: SpaceDict,
+    ):
+        size = self._size
+        observation_space = copy.deepcopy(observation_space)
+        if size:
+            for key in observation_space.spaces:
+                if (
+                    key in self.trans_keys
+                    and observation_space.spaces[key].shape[-3:-1] != size
+                ):
+                    h, w = get_image_height_width(
+                        observation_space.spaces[key], channels_last=True
+                    )
+                    logger.info(
+                        "Center cropping observation size of %s from %s to %s"
+                        % (key, (h, w), size)
+                    )
+
+                    observation_space.spaces[key] = overwrite_gym_box_shape(
+                        observation_space.spaces[key], size
+                    )
+        return observation_space
+
+    def _transform_obs(self, obs: torch.Tensor) -> torch.Tensor:
+        return center_crop(
+            obs,
+            self._size,
+            channels_last=self.channels_last,
+        )
+
+    @torch.no_grad()
+    def forward(
+        self, observations: Dict[str, torch.Tensor]
+    ) -> Dict[str, torch.Tensor]:
+        if self._size is not None:
+            observations.update(
+                {
+                    sensor: self._transform_obs(observations[sensor])
+                    for sensor in self.trans_keys
+                    if sensor in observations
+                }
+            )
+        return observations
+
+    @classmethod
+    def from_config(cls, config: Config):
+        return cls(
+            (
+                config.RL.POLICY.OBS_TRANSFORMS.CENTER_CROPPER.HEIGHT,
+                config.RL.POLICY.OBS_TRANSFORMS.CENTER_CROPPER.WIDTH,
+            )
+        )
+
+
+class Cube2Equirec(nn.Module):
+    def __init__(self, equ_h, equ_w, cube_length, CUDA=False):
```

Comment:
CUDA really shouldn't be an arg to `__init__` IMO.  You can do 
```
self.grid  = self.grid.to(batch.device)
self.orientation_mask = self.orientation_mask.to(batch.device)
```
in forward and that will be a no-op after the first call.